cvmfs/json_document_write.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_JSON_DOCUMENT_WRITE_H_
+#define CVMFS_JSON_DOCUMENT_WRITE_H_
+
+#include <string>
+#include <vector>
+
+// This class is used for marshalling JSON objects to strings.
+// When adding an object, use quoted = true for strings and
+// quoted = false for numbers, nested objects, etc.
+
+class JsonStringGenerator {
+  struct JsonStringEntry {
+    std::string key;
+    std::string val;
+    bool quoted;
+
+    JsonStringEntry() {
+      quoted = true;
+    }
+  };
+
+ public:
+  void PushBack(std::string key, std::string val, bool quoted = true) {
```

Comment:
I think I would prefer two public interfaces: `AddQuoted` and `AddUnquoted`, otherwise one always needs to lookup what the `true` or `false` actually means.

Reply:
Hummm honestly I didn't write this code. Changing this will require to change all the invocation.

If we really want to go that route, the I would prefer to have a PushBack overloaded, one for string, one for int, one for float, one for boolean and one for null, leaving the management of array and object to the caller.
Anyway, this already grow to much so I will just push back for string and numbers, leaving the implementation of bools and null to when it will be necessary.

Reply:
Nah this is a bad idea within our constraints.